crates/ibc/src/applications/transfer/relay/on_recv_packet.rs

```diff
@@ -60,3 +60,78 @@ pub fn process_recv_packet<Ctx: 'static + TokenTransferContext>(
 
     Ok(())
 }
+
+#[cfg(feature = "val_exec_ctx")]
+pub use val_exec_ctx::*;
+#[cfg(feature = "val_exec_ctx")]
+mod val_exec_ctx {
+    pub use super::*;
+    use crate::core::ics04_channel::handler::ModuleExtras;
+
+    pub fn process_recv_packet_execute<Ctx: TokenTransferContext>(
+        ctx: &mut Ctx,
+        packet: &Packet,
+        data: PacketData,
+    ) -> Result<ModuleExtras, (ModuleExtras, TokenTransferError)> {
```

Comment:
> Are we trying to move away from the output argument?

Yes.

> A logger should work till the last line of code is worked, but it's not true for events. Hence, the logger handler should be passed to functions somehow similar to the output arg.

Actually the events also need to be returned even in the case of an error (see [ibc-go](https://github.com/cosmos/ibc-go/blob/c0acd5bd1778f2b7ecdf593006f56bd3e273bd49/modules/apps/transfer/ibc_module.go#L176-L210)).

This interface looks funny because I wanted to still use `?` for errors, but then `ModuleExtras` need to be returned also in the case of an error. `(ModuleExtras, Result<(), Error>)` would not have allowed the use of `?`.

There's probably a cleaner way to do it, but again, I prioritized correctness.